lib/src/rules/cascade_invocations.dart

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library linter.src.rules.cascade_invocations;
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:linter/src/linter.dart';
+
+const _desc = r'Cascade consecutive method invocations on the same reference.';
+
+const _details = r'''
+
+**DO** Use the cascading style when succesively invoking methods on the same
+ reference.
+
+**BAD:**
+```
+SomeClass someReference = new SomeClass();
+someReference.firstMethod();
+someReference.secondMethod();
+```
+
+**BAD:**
+```
+SomeClass someReference = new SomeClass();
+...
+someReference.firstMethod();
+someReference.secondMethod();
+```
+
+**GOOD:**
+```
+SomeClass someReference = new SomeClass()
+    ..firstMethod()
+    ..secondMethod();
+```
+
+**GOOD:**
+```
+SomeClass someReference = new SomeClass();
+...
+someReference
+    ..firstMethod()
+    ..secondMethod();
+```
+
+''';
+
+SimpleIdentifier _findTarget(AstNode node) {
+  Expression expression;
+  if (node is ExpressionStatement) {
+    expression = node.expression;
+  }
+
+  if (expression is MethodInvocation &&
+      expression.realTarget is SimpleIdentifier) {
+    return expression.realTarget;
+  }
+
+  if (expression is PrefixedIdentifier &&
+      expression.prefix is SimpleIdentifier) {
+    return expression.prefix;
+  }
+
+  return null;
+}
+
+/// Rule to lint consecutive invocations of methods or getters on the same
+/// reference that could be done with the cascade operator.
+class CascadeInvocations extends LintRule {
+  _Visitor _visitor;
+
+  /// Default constructor.
+  CascadeInvocations()
+      : super(
+            name: 'cascade_invocations',
+            description: _desc,
+            details: _details,
+            group: Group.style) {
+    _visitor = new _Visitor(this);
+  }
+
+  @override
+  AstVisitor getVisitor() => _visitor;
+}
+
+class _Visitor extends SimpleAstVisitor {
+  final LintRule rule;
+  _Visitor(this.rule);
+
+  @override
+  void visitExpressionStatement(ExpressionStatement node) {
+    final prefixIdentifier = _findTarget(node);
+    if (prefixIdentifier == null) {
+      return;
+    }
+
+    final Block body = node.getAncestor((n) => n is Block);
+    final previousNodes = body.childEntities.takeWhile((n) => n != node);
```

Comment:
If you make the change above, then you can look at the statements in the block rather than the childEntities.